Make bioswale polygons on map pages clickable

maurizi opened the pull request:
I explored using multiple UTF grids but ran into several issues:
 - clicks do not "fall through" to the lower UTF grid
 - max/min zoom properties are not respected

Instead, I implemented a WMS-style server backend for determining if a
polygon is present at a given lat/lng.  UTF grids are still used for point
based map features, and no network requests are made if the map does not
support polygonal map features.

Connects to #2114

jwalgran commented on opentreemap/treemap/js/src/MapManager.js:
```diff
@@ -58,6 +61,27 @@ MapManager.prototype = {
+                map.utfEvents = Bacon.mergeAll(
+                    plotUtfEventStream,
+                    polygonDataStream
+                ).log();
```
Leftover ``log`` statement from debugging.

jwalgran commented on opentreemap/stormwater/views.py:
```diff
@@ -0,0 +1,34 @@
+    if len(features) > 0:
+        # This is currently only being used to complement UTF grid plot data,
+        # so the structure of the response is designed to mirror the utf grid
+        return {
```
I like the idea of matching the UTF grid schema so that the downstream handler does not have to care.

jwalgran commented on opentreemap/stormwater/views.py:
```diff
@@ -0,0 +1,34 @@
+
+    features = PolygonalMapFeature.objects.distance(point)\
+        .filter(instance=instance)\
+        .filter(polygon__distance_lte=(point, D(m=settings.MAP_CLICK_RADIUS)))\
```
I don't see MAP_CLICK_RADIUS defined as part of this PR.

In general, a click buffer is most effectively defined as a function of the current zoom level, since the distance-to-pixel ratio is cut in half each time you zoom in. Because we are only dealing with levels 15 through 19, it may be possible to choose a reasonable, single value.

jwalgran replied:
Nevermind, this reused the radius parameter established by the API

jwalgran commented:
Tested. Trees within the bounds of a polygonal map feature are still clickable.

The functionality is good, but I think we should address the radius issue before we ship it.

maurizi commented:
@jwalgran I've removed the stray log statement, and I added a `distance` parameter, which the client halves for every zoom level you go in.  Please take another look.

jwalgran commented:
+1 (did not retest)
